lib/reform/form/active_model.rb

```diff
@@ -92,7 +92,17 @@ def model_name
   private
     def active_model_name_for(string)
       return ::ActiveModel::Name.new(OpenStruct.new(:name => string)) if Reform.rails3_0?
-      ::ActiveModel::Name.new(self, nil, string)
+      ::ActiveModel::Name.new(self, determine_namespace(string) , string)
+    end
+
+    def determine_namespace(string)
+      begin
+        string.constantize.parents.detect do |n|
+          n.respond_to?(:use_relative_model_naming?) && n.use_relative_model_naming?
+        end
+      rescue NameError
```

Comment:
This is a safe guarantee incase the string passed-in can not be constantize. Any good solution for this?